src/patch/apply.js

```diff
@@ -42,7 +39,30 @@ export function applyPatch(source, uniDiff, options = {}) {
             return false;
           }
         }
+        toPos++;
       }
+    }
+
+    return true;
+  }
+
+ // Sanity check the input string. Fail if we don't match.
+ for (let i = 0; i < hunks.length; i++) {
+    let hunk = hunks[i];
+
+    if (!hunkFits(hunk, hunk.oldStart - 1)) {
+      return false;
```

Comment:
This is WiP to allow to calculate the location offset where the hunks best fit. This is to fix issue #84